### DISCUSSION TITLE
Proposal: merge mode specified in override file

### DISCUSSION THREAD

**Discussion by Patrick8639** [dotnet/docfx#9908] [Ideas]
### The problem
Sometimes, I need to **add** documentation after the remarks for a class (or a member). Currently, when creating an override file for the remarks, docfx just replaces the remarks with the override content.

### A solution
The yaml header for an override has currently the following format:
```yaml
---
uid: <uid>
remarks: *content
---
```
We can add a new parameter in this header, in the following form `placement: <value>`. Where `value` can be `after`, `before` or `replace`.
If the parameter is defined and the value is one of the three valid ones, the content is updated accordingly. Otherwise, the old merge algorithm is used, so there is no breaking change in existing documentation.

### To do
1. In `Docfx.Common`, define a `IModelWithMetada` interface, for a single property: `Metadata`, of type `Dictionary<string, object>`.
2. Make `Docfx.Dotnet.ItemViewModel` implements `IModelWithMetadata` (it already define the property).
3. In `Docfx.Common.ReflectionEntityMerger.Merge`, detect and manage the parameter.

I already have a working version, but it needs to be polished. Before I finish and make a pull request, I would like to know what you think of this solution.

PS: English is not my mother tongue, so, if you have better names, please tell me. Also, excuse the English errors.

**Comment by filzrev** [comment]
There is a `Feature Request` issue in #7648.
I thought it would make sense to have an 'add content' options for overwrite document rather than 'replace' all the content.

**Comment by yufeih** [comment]
What if there are two properties to override where one wants `replace` and the other wants `append`?

**Comment by Patrick8639** [comment]
> What if there are two properties to override where one wants replace and the other wants append?

In the header of an overwrite file, there is the name of the property, so the placement is specified for this property:

```yaml
---
uid: namespace.class.member
remarks: *content
placement: after
---
```
